src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpFixAllSpanMappingService.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeFixes
+{
+    [ExportLanguageService(typeof(IFixAllSpanMappingService), LanguageNames.CSharp), Shared]
+    internal sealed class CSharpFixAllSpanMappingService : AbstractFixAllSpanMappingService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpFixAllSpanMappingService()
+        {
+        }
+
+        protected override async Task<ImmutableDictionary<Document, ImmutableArray<TextSpan>>> GetFixAllSpansIfWithinGlobalStatementAsync(
+            Document document, TextSpan diagnosticSpan, FixAllScope fixAllScope, CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(fixAllScope is FixAllScope.ContainingMember or FixAllScope.ContainingType);
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnosticSpan);
+            if (node.GetAncestorOrThis<GlobalStatementSyntax>() is null)
+                return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty;
+
+            // Compute the fix all span for the global statements to be fixed.
+            // If the file has non-global statements towards the end, they need to be excluded
+            // from the fix all span.
+            var fixAllSpan = root.FullSpan;
+            var firstNonGlobalStatement = root.ChildNodes().FirstOrDefault(n => n is not GlobalStatementSyntax);
+            if (firstNonGlobalStatement is not null)
+            {
+                // Bail out for compiler error case where a non-global statement precedes a global statement.
+                // C# compiler requires all global statements to preceed non-global statements.
+                var globalStatements = root.ChildNodes().OfType<GlobalStatementSyntax>();
+                if (globalStatements.Any(g => firstNonGlobalStatement.SpanStart < g.SpanStart))
+                    return ImmutableDictionary<Document, ImmutableArray<TextSpan>>.Empty;
```

Comment:
this doesn't sppear fixed.  your `root.ChildNodes().FirstOrDefault(n => n is not GlobalStatementSyntax)` will return things like a `using directive`.  That using directive is allowed to come before a normal `global statement`.  but this will return empty here.

Reply:
Thanks - I updated the check to now use [SyntaxFacts.IsNamespaceMemberDeclaration](https://sourceroslyn.io/Microsoft.CodeAnalysis.CSharp/R/9217de0879160b70.html), which matches the logic used by the language parser to report this error: https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Parser/LanguageParser.cs#L604-L628